packages/indexer-agent/src/db/migrations/11-update-deployment-names.ts

```diff
@@ -0,0 +1,129 @@
+import { Logger } from '@graphprotocol/common-ts'
+import {
+  formatDeploymentName,
+  indexerError,
+  IndexerErrorCode,
+  IndexingStatusResolver,
+  NetworkSubgraph,
+  SubgraphDeployment,
+} from '@graphprotocol/indexer-common'
+import { Client } from 'jayson/promise'
+import pMap from 'p-map'
+
+interface MigrationContext {
+  logger: Logger
+  indexingStatusResolver: IndexingStatusResolver
+  graphNodeAdminEndpoint: string
+  networkSubgraph: NetworkSubgraph
+}
+
+interface Context {
+  context: MigrationContext
+}
+
+interface SubgraphRedeployment {
+  newName: string
+  ipfsHash: string
+  nodeId: string
+}
+
+export async function up({ context }: Context): Promise<void> {
+  const { logger } = context
+
+  // Get a list of all currently deployed sugraphs.
+  logger.debug('Fetching active deployments')
+  const currentDeploymentDetails =
+    await context.indexingStatusResolver.subgraphDeploymentsDetails()
+
+  if (currentDeploymentDetails.length === 0) {
+    logger.info('No active deployments found. Migration not necessary.')
+  }
+
+  // Query the Network Subgraph for more details about those subgraph deployments.
+  // It's OK to ignore missing subgraph deployments if they are not found on chain.
+  logger.debug('Query Network Subgraph for deployment details')
+  const subgraphDeployments: SubgraphDeployment[] = (
+    await pMap(
+      currentDeploymentDetails.map(id => id.id.ipfsHash),
+      sgd => context.networkSubgraph.subgraphDeployment(sgd),
+    )
+  ).filter((sgd): sgd is SubgraphDeployment => Boolean(sgd))
+
+  if (subgraphDeployments.length === 0) {
+    logger.info(
+      'No subgraph deployments worth redeploying. Migration not necessary.',
+    )
+  }
+
+  // Produce the updated subgraph deployment names. To do that, we need merged data that comes from
+  // Graph Node (nodeID) and the Network Subgraph (subgraph owner and name).
+  const subgraphRedeployments: SubgraphRedeployment[] = []
+  for (const sgd of subgraphDeployments) {
+    const details = currentDeploymentDetails.find(
+      current => current.id.ipfsHash === sgd.id.ipfsHash,
```

Comment:
I have updated the variable names to distinguish between assignments and subgraph details (found in the Network) in c481f72.

Reply:
As of now, the `SubgraphRedeployment` interface requires information from both the current assignment (nodeID) and the Network (subgraph and creator names). This interface exists only in this migration and represents all the required fields to issue a `subgraph_deployment` RPC over Graph Node. 

I tried to split the preparation of such data into three steps:

1. In the [first part](https://github.com/graphprotocol/indexer/blob/b5092a0bc8815b6afc4dd39e4cd16c12d29454e7/packages/indexer-agent/src/db/migrations/11-update-deployment-names.ts#L33-L40) of the code, we fetch all the current assignments. _(their `nodeIds`)_ 
2. In the [second part](https://github.com/graphprotocol/indexer/blob/b5092a0bc8815b6afc4dd39e4cd16c12d29454e7/packages/indexer-agent/src/db/migrations/11-update-deployment-names.ts#L42-L58), we query the Network to supplement the data for those assignments. _(subgraph and creator names)_
3. Finally, in the [third step](https://github.com/graphprotocol/indexer/blob/b5092a0bc8815b6afc4dd39e4cd16c12d29454e7/packages/indexer-agent/src/db/migrations/11-update-deployment-names.ts#L60-L79), we attempt to produce such a `SubgraphRedeployment` object, combining information from the two previous steps.

Reply:
For the `formatDeploymentName` function to work with just the IPFS hash as a parameter, it would need to have the Network Subgraph as a dependency because the new name scheme requires data gathered from it.

I believe the Network data is already present in some scopes where we are using `formatDeploymentName`, so we should take measures to avoid querying unnecessarily.

Reply:
I've refactored the core migration logic in 8187cb1.